app_users/models.py

```diff
@@ -233,6 +234,16 @@ def get_or_create_personal_workspace(self) -> tuple["Workspace", bool]:
 
         return Workspace.objects.get_or_create_from_user(self)
 
+    @cached_property
```

Comment:
django has its own `cached_property` decorator for use with models https://docs.djangoproject.com/en/5.1/ref/utils/#django.utils.functional.cached_property

Reply:
difference here: https://stackoverflow.com/a/68594071

we want thread safety and work in python <3.12, django's is more efficient there